### ISSUE TITLE
Component is not re-rendering in some case

### ISSUE THREAD

**Issue by sqal** [preactjs/preact#3677]
- [ ] Check if updating to the latest Preact version resolves the issue

**Describe the bug**
Component is not re-rendering on state change. Seems like this bug was introduced in the version 10.10.3.

**To Reproduce**

https://codesandbox.io/s/serene-tamas-vmudmq?file=/src/index.js

Wait 1 second and click on "hide me" button.

**Expected behavior**
Button should should hide when clicked, however nothing happens.  This is a strange bug, because if you swap order of the declared variables, then it works correctly

```js
const [show, setShow] = useState(false);
const [step, setStep] = useState(Step.LOADING);
```

**Comment by JoviDeCroock** [comment]
Should be fixed by #3765

**Comment by JoviDeCroock** [comment]
Can confirm works in 10.10.4 https://codesandbox.io/s/nifty-germain-y91wie?file=/package.json
